Update to latest Coursier 2.1.2

## problem

sbt-coursier PR to update to Coursier 2.1.2 is blocked https://github.com/coursier/sbt-coursier/pull/451:

```bash
[error] Unrecognized: darwin/aarch64/libzstd-jni-1.5.4-2.dylib
[error] Unrecognized: darwin/x86_64/libzstd-jni-1.5.4-2.dylib
[error] Unrecognized: freebsd/amd64/libzstd-jni-1.5.4-2.so
[error] Unrecognized: freebsd/i386/libzstd-jni-1.5.4-2.so
[error] Unrecognized: linux/aarch64/libzstd-jni-1.5.4-2.so
[error] Unrecognized: linux/amd64/libzstd-jni-1.5.4-2.so
[error] Unrecognized: linux/arm/libzstd-jni-1.5.4-2.so
[error] Unrecognized: linux/i386/libzstd-jni-1.5.4-2.so
[error] Unrecognized: linux/loongarch64/libzstd-jni-1.5.4-2.so
[error] Unrecognized: linux/mips64/libzstd-jni-1.5.4-2.so
[error] Unrecognized: linux/ppc64/libzstd-jni-1.5.4-2.so
[error] Unrecognized: linux/ppc64le/libzstd-jni-1.5.4-2.so
[error] Unrecognized: linux/s390x/libzstd-jni-1.5.4-2.so
[error] Unrecognized: win/amd64/libzstd-jni-1.5.4-2.dll
[error] Unrecognized: win/x86/libzstd-jni-1.5.4-2.dll
[error] java.lang.AssertionError: assertion failed
[error] 	at scala.Predef$.assert(Predef.scala:208)
[error] 	at coursier.ShadingPlugin$.onlyNamespaces(ShadingPlugin.scala:134)
[error] 	at coursier.ShadingPlugin$.$anonfun$projectSettings$28(ShadingPlugin.scala:220)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2[219](https://github.com/coursier/sbt-coursier/actions/runs/4744305295/jobs/8425036011?pr=451#step:6:220)$1(TypeFunctions.scala:62)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error] 	at sbt.Execute.work(Execute.scala:291)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error] 	at java.lang.Thread.run(Thread.java:750)
[error] (lm-coursier-shaded / shadedPackageBin) java.lang.AssertionError: assertion failed
```

## expectation

We should keep up with Coursier versions, probably.

## notes

See also https://github.com/coursier/coursier/issues/2749

--- a/project/Dependencies.scala
+++ b/project/Dependencies.scala
@@ -77,7 +77,7 @@ object Dependencies {
   def addSbtZincCompile = addSbtModule(sbtZincPath, "zincCompile", zincCompile)
   def addSbtZincCompileCore = addSbtModule(sbtZincPath, "zincCompileCore", zincCompileCore)
 
-  val lmCoursierShaded = "io.get-coursier" %% "lm-coursier-shaded" % "2.0.16"
+  val lmCoursierShaded = "io.get-coursier" %% "lm-coursier-shaded" % "2.1.0"
 
   def sjsonNew(n: String) =
     Def.setting("com.eed3si9n" %% n % "0.9.1") // contrabandSjsonNewVersion.value

